Library/Homebrew/test/livecheck/strategy/sparkle_spec.rb

```diff
@@ -7,91 +7,137 @@
-          <language>en</language>
-          #{first_item}
-          #{second_item}
-        </channel>
-      </rss>
+    v120_item_with_macos_os = <<~EOS
+      <item>
+        <title>#{item_hashes[:v120][:title]}</title>
+        <sparkle:minimumSystemVersion>10.10</sparkle:minimumSystemVersion>
+        <sparkle:releaseNotesLink>https://www.example.com/example/#{item_hashes[:v120][:short_version]}.html</sparkle:releaseNotesLink>
+        <pubDate>#{item_hashes[:v120][:pub_date]}</pubDate>
+        <enclosure os="macos" url="#{item_hashes[:v120][:url]}" sparkle:shortVersionString="#{item_hashes[:v120][:short_version]}" sparkle:version="#{item_hashes[:v120][:version]}" length="12345678" type="application/octet-stream" sparkle:dsaSignature="ABCDEF+GHIJKLMNOPQRSTUVWXYZab/cdefghijklmnopqrst/uvwxyz1234567==" />
+      </item>
     EOS
 
-    omitted_items = appcast.sub("</item>", "</item>\n#{items_to_omit}")
-    beta_channel_item = appcast.sub(
-      first_item,
-      first_item.sub(
-        "</title",
-        "</title>\n<sparkle:channel>beta</sparkle:channel>",
-      ),
+    # This main `appcast` data is intended as a relatively normal example.
+    # As such, it also serves as a base for some other test data.
+    appcast = create_appcast_xml <<~EOS
+      #{v123_item}
+      #{v122_item}
+      #{v121_item_with_osx_os}
+      #{v120_item_with_macos_os}
+    EOS
+
+    omitted_items = create_appcast_xml <<~EOS
+      #{v123_item.sub(%r{<(enclosure[^>]+?)\s*?/>}, '<\1 os="not-osx-or-macos" />')}
+      #{v123_item.sub(/(<sparkle:minimumSystemVersion>)[^<]+?</m, '\1100<')}
+      <item>
+      </item>
+    EOS
+
+    beta_channel_item_string = v123_item.sub(
+      /<sparkle/, "<sparkle:channel>beta</sparkle:channel><sparkle"
     )
-    no_versions_item =
-      appcast
-        .sub(second_item, "")
-        .gsub(/sparkle:(shortVersionString|version)="[^"]+?"\s*/, "")
-        .sub(
-          "<title>#{item_hash[0][:title]}</title>",
-          "<title>Version</title>",
-        )
-    no_items = appcast.sub(%r{<item>.+</item>}m, "")
-    undefined_namespace = appcast.sub(/\s*xmlns:sparkle="[^"]+?"/, "")
+
+    beta_channel_item = appcast.sub(v123_item, beta_channel_item_string)
+
+    no_versions_item = create_appcast_xml <<~EOS
+      <item>
+        <title>Version</title>
+        <sparkle:minimumSystemVersion>10.10</sparkle:minimumSystemVersion>
+        <sparkle:releaseNotesLink>https://www.example.com/example/#{item_hashes[:v123][:short_version]}.html</sparkle:releaseNotesLink>
+        <pubDate>#{item_hashes[:v123][:pub_date]}</pubDate>
+        <enclosure url="#{item_hashes[:v123][:url]}" length="12345678" type="application/octet-stream" sparkle:dsaSignature="ABCDEF+GHIJKLMNOPQRSTUVWXYZab/cdefghijklmnopqrst/uvwxyz1234567==" />
+      </item>
+    EOS
+
+    no_items = create_appcast_xml
+
+    undefined_namespace = appcast.sub(/xmlns:sparkle="[^"]+"/, "")
```

Comment:
```suggestion
    undefined_namespace = appcast.sub(/\s*xmlns:sparkle="[^"]+"/, "")
```

This is just a small tweak to reinstate part of the existing regex. Namely, we want to ensure that the substitution doesn't leave behind any leading whitespace. It *shouldn't* make a functional difference but better safe than sorry.